### PR TITLE
Scir lsf

### PR DESCRIPTION
**Pull request by juanprietob**

### CODE REVIEW

File: BrainPortal/lib/scir_lsf.rb [hunk 1]
```diff
@@ -28,73 +28,62 @@ class ScirLsf < Scir
...
 
-    def update_job_info_cache #:nodoc:
-      # FIXME: modify this method based
+    def update_job_info_cache #:nodoc:      
       # on the implementation of bstat.  This method is supposed to
       # list all the statuses of the running tasks, and to keep it in
       # the @job_info_cache array.
-      out, err = bash_this_and_capture_out_err("bstat -f -u #{CBRAIN::Rails_UserName.to_s.bash_escape}")
-      raise "Cannot get output of 'bstat -f' ?!?" if out.blank? && ! err.blank?
+      out, err = bash_this_and_capture_out_err("bjobs -a -noheader -J #{shell_escape(self.name)}")
```
Comment (glatard) [inline BrainPortal/lib/scir_lsf.rb]: -J should be removed. The following option should be added: -u #{CBRAIN::Rails_UserName.to_s.bash_escape}

File: BrainPortal/lib/scir_lsf.rb [hunk 2]
```diff
@@ -116,25 +105,28 @@ def qsubout_to_jid(txt) #:nodoc:
...
-      command += "-j oe "                           if self.join
+      script = self.wd + '/' +self.arg[0]
+      stdout = self.stdout.sub(':', '') if self.stdout
+      stderr = self.stderr.sub(':', '') if self.stderr
+
+      File.chmod(0755, script)
+
+      command  = "bsub "      
+      command += "-J #{shell_escape(self.name)} "   if self.name
+      command += "-u #{shell_escape(self.name)} "   if self.name
```
Comment (glatard) [inline BrainPortal/lib/scir_lsf.rb]: Remove -u option.

File: BrainPortal/lib/scir_lsf.rb [hunk 3]
```diff
@@ -116,25 +105,28 @@ def qsubout_to_jid(txt) #:nodoc:
...
+      command += "-cwd #{shell_escape(self.wd)} "     if self.wd
+      command += "-o #{shell_escape(stdout)} " if stdout
+      command += "-e #{shell_escape(stderr)} " if stderr 
       command += "-q #{shell_escape(self.queue)} "  unless self.queue.blank?
       command += "#{Scir.cbrain_config[:extra_qsub_args]} " unless Scir.cbrain_config[:extra_qsub_args].blank?
-      command += "#{self.tc_extra_qsub_args} "              unless self.tc_extra_qsub_args.blank?
-      command += "-l walltime=#{self.walltime.to_i} "       unless self.walltime.blank?
-      command += "#{shell_escape(self.arg[0])}"
```
Comment (glatard) [inline BrainPortal/lib/scir_lsf.rb]: Add -W #{self.walltime.to_i}

File: BrainPortal/lib/scir_lsf.rb [hunk 4]
```diff
@@ -116,25 +105,28 @@ def qsubout_to_jid(txt) #:nodoc:
...
 
-      command  = "bsub "
-      command += "-S /bin/bash "                    # Always
-      command += "-r n "                            # Always
-      command += "-d #{shell_escape(self.wd)} "     if self.wd 
-      command += "-N #{shell_escape(self.name)} "   if self.name
-      command += "-o #{shell_escape(self.stdout)} " if self.stdout
-      command += "-e #{shell_escape(self.stderr)} " if self.stderr
-      command += "-j oe "                           if self.join
+      script = self.wd + '/' +self.arg[0]
```
Comment (glatard) [inline BrainPortal/lib/scir_lsf.rb:116]: Really necessary?

File: BrainPortal/lib/scir_lsf.rb [hunk 5]
```diff
@@ -28,73 +28,62 @@ class ScirLsf < Scir
...
-            jid = Regexp.last_match[1]
-          end
-          next
-        end
-        next unless line =~ /^\s*job_state\s*=\s*(\S+)/i
-        state = statestring_to_stateconst(Regexp.last_match[1])
-        @job_info_cache[jid.to_s] = { :drmaa_state => state }
+        bjob_stat = line.gsub(/\s+/m, ' ').strip.split(" ")
+        jid = bjob_stat[0]
+        stat = bjob_stat[2]
```
Comment (glatard) [inline BrainPortal/lib/scir_lsf.rb:42]: Add: `stat = statestring_to_stateconst(stat)`

Comment (juanprietob) [inline BrainPortal/lib/scir_lsf.rb:42]: I added this line
I have this as output: in the variable @job_info_cache

`ext-ruby-2.2.3-p173 :102 > print @job_info_cache
{"846007"=>{:drmaa_state=>"STATE_DONE"}, "846264"=>{:drmaa_state=>"STATE_DONE"}, "846276"=>{:drmaa_state=>"STATE_DONE"}} => nil`

However, the state in cbrain portal does not get refreshed accordingly.

Comment (glatard) [inline BrainPortal/lib/scir_lsf.rb:42]: The statestring_to_stateconst method should not return STATE_DONE when the job is done. In fact, CBRAIN will detect that the job is done when it doesn't appear in the bsub output any more. If a finished job still appears in the output of bsub, then it should be detected in this method and such a job shouldn't be added to @job_info_cache. If you prefer to _not list_ the finished jobs, which will also speedup parsing, then you should remove option '-a' in the bjobs command.

Comment (juanprietob) [inline BrainPortal/lib/scir_lsf.rb:42]: In any case, after a while `the bjobs` command returns `no jobs found`. Nevertheless, the status is not updated.

Comment (glatard) [inline BrainPortal/lib/scir_lsf.rb:42]: It really should be... did you check that @job_info_cache doesn't contain the id of completed jobs?

**Pull request merged**